File: tests/eval_c/510.c
#include <assert.h>
#include <string.h>

void f510(const char *s) {
    assert(strcmp(s, "abc") == 0 || strcmp(s, "ab") == 0 || strcmp(s, "a") == 0 ||
           strcmp(s, "ac") == 0 || strcmp(s, "bc") == 0 || strcmp(s, "b") == 0 ||
           strcmp(s, "c") == 0 || strcmp(s, "") == 0);
}

File: benchmarks/oopsla25/subjects/510.py
#!/usr/bin/env python3

import sys

def f510(s: str):
  assert s == "abc" or s == "ab" or s == "a" or s == "ac" or s == "bc" or s == "b" or s == "c" or s == ""

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f510(my_string)
